### PR TITLE
Clarify the security implication of using this module in the README

### PR DESCRIPTION
**Pull request by mcollina**
As title, let's make it clearer what the responsibilities of this plugin are.

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by salmanm — APPROVED** [review]

**Review by simoneb — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -3,10 +3,20 @@
 ![Node.js CI](https://github.com/fastify/fastify-csrf/workflows/Node.js%20CI/badge.svg)
 [![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat)](http://standardjs.com/)
 
-A plugin for adding [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery) protection to Fastify.
-If you want to learn more about CSRF, see [pillarjs/understanding-csrf](https://github.com/pillarjs/understanding-csrf) and [Cross-Site Request Forgery Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html)
+This plugin helps developers protect their fastify server against [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery) attacks.
+In order to fully protect against CSRF, developers should study [Cross-Site Request Forgery Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html)
+in great details. See also [pillarjs/understanding-csrf](https://github.com/pillarjs/understanding-csrf) as a good guide.
```
Comment (jsumners) [inline README.md:8]:
```suggestion
in depth. See also [pillarjs/understanding-csrf](https://github.com/pillarjs/understanding-csrf) as a good guide.
```

File: README.md [hunk 2]
```diff
@@ -3,10 +3,20 @@
...
+## Security Disclaimer
+
+Securing applications against CSRF is a _developer responsibility_ and it should not be fully trusted to any third party modules.
+We do not claim this this module is able to protect an application without a clear study of CSRF, its impact and the needed mitigations.
+fastify-csrf provides a series of utilities that developers can use to secure their application.
+We recommend using [fastify-helmet](https://github.com/fastify/fastify-helmet) to implement some of those mitigations.
+
+Security is always a tradeoff between risk mitigation, functionality and developer experience.
+As a result we will not consider a security vulnerability a default that might be unsafe in certain scenarios,
+as long as this module provides a way to provide a full mitigation.
```
Comment (jsumners) [inline README.md:19]:
```suggestion
Security is always a tradeoff between risk mitigation, functionality and developer experience.
As a result we will not consider a report of a plugin default configuration option as security
vulnerability that might be unsafe in certain scenarios as long as this module provides a
way to provide full mitigation through configuration.
```

File: README.md [hunk 3]
```diff
@@ -155,7 +121,20 @@ fastify.route({
 })
 ```
 
-### Options
+### Securing the secret
+
+The `secret` shown in the code above is strictly just a sample. In all cases, you'd need to make sure that the `secret` is:
```
Comment (jsumners) [inline README.md:126]:
```suggestion
The `secret` shown in the code above is strictly just an example. In all cases, you'd need to make sure that the `secret` is:
```

File: README.md [hunk 4]
```diff
@@ -155,7 +121,20 @@ fastify.route({
...
+- **Never** hard-coded in the code or `.env` files or anywhere in the repository
+- Stored in some external services like KMS, Vault or something similar
+- Read at run-time and supplied in this option
+- Long enough
```
Comment (jsumners) [inline README.md:130]:
```suggestion
- Of significant character length to provide adequate entropy
```

**Review by jsumners — APPROVED** [review]
LGTM with a few tweaks as suggested

**Pull request merged**